File: ITP1/ITP1_7_D.py
matrixA = []
matrixB = []
n, m, l = map(int, input().split())

#matrixCの初期化(n行l列)
matrixC = [[0 for i in range (l)]for k in range (n)]

#matrixAの入力
for i in range (n):
    matrixA.append(list(map(int, input().split())))
#matrixBの入力
for j in range (m):
    matrixB.append(list(map(int, input().split())))

#matrixCの導出
for i in range (n):
    for j in range (m):
        for k in range (l):
            matrixC[i][k] += matrixA[i][j] * matrixB[j][k]

#matrixCの出力
for i in range (n):
    print(" ".join(map(str, matrixC[i])))

File: ITP1/ITP1_7_D.c
#include <stdio.h>
int main(){
  int n, m, l;
  int i, j, k;
  long input;
  long a[200][200];
  long b[200][200];
  long c[200][200];

  //input n, m, l
  scanf("%d %d %d", &n, &m, &l);

  //input matrixA and matrixB
  for(i=0; i<n; i++){
    for(j=0; j<m; j++){
      scanf("%ld", &input);
      a[i][j]=input;
    }
  }
  for(j=0; j<m; j++){
    for(k=0; k<l; k++){
      scanf("%ld", &input);
      b[j][k]=input;
    }
  }

  //calcuration (matrixA * matrixB)
  for(i=0; i<n; i++){
    for(j=0; j<m; j++){
      for(k=0; k<l; k++){
        c[i][k]+=a[i][j]*b[j][k];
      }
    }
  }

  //output matrixC
  for(i=0; i<n; i++){
    for(k=0; k<l; k++){
      if(k != l){
        printf("%ld ", c[i][k]);
      }else{
        printf("%ld", c[i][k]);
      }
    }
    printf("\n");
  }
  return 0;
}
